RELEASES.md

```diff
@@ -99,6 +99,10 @@ v4.8.0 (development in progress)
   and `#guard_msgs (ordering := sorted) in cmd` sorts the messages in lexicographic order before checking.
   PR [#3883](https://github.com/leanprover/lean4/pull/3883).
 
+* The `#guard_msgs` command now supports showing a diff between the expected and actual outputs. This feature is currently
+  disabled by default, but can be enabled with `set_option guard_msgs.diff true`. Depending on user feedback, this option
+  may default to `true` in a future version of Lean.
```

Comment:
Why not on by default? If it's off I expect few people will ever benefit from it, simply because they don't know about it.

We can still turn it off by default if it's not well received.

Reply:
For both this and in expected type mismatch messages, I think both views are useful, but I don't think a `set_option` is the best way to surface it to users, a toggle button in the info view would be better.

Reply:
I agree that an infoview toggle would be better, but it's not achievable in the time I have right now.

Another reason to have it off by default right now is that the practicality of something like this is hard to assess in the abstract, or with contrived examples. I plan to have it on for a couple of months, and I'm sure a bunch of little papercuts will surface that I can fix.

I'm already expecting that I'll want a threshold of similarity that triggers the diff rendering (e.g. if less than X% of lines are edited, show a diff, otherwise don't), but I'd rather base that kind of thing on real experience by myself and others.

The plan is to make it on by default once it makes people happy enough, but I'd rather have the experiment be opt-in to start with.

Reply:
Fair enough!